code/modules/ruins/lavalandruin_code/tomb.dm

```diff
@@ -0,0 +1,363 @@
+	..()
+
+//todo hit
+
+/mob/living/simple_animal/hostile/cryptguard/Move()
+	playsound(src,'sound/effects/stonedoorfast.ogg',40,1)
+	..()
+
+/mob/living/simple_animal/hostile/cryptguard/proc/awaken()
+	woke = TRUE
+	if(!istype(src, /mob/living/simple_animal/hostile/cryptguard/leader))
+		name = "crypt guardian"
+		icon_state = "crypt-1"
+	toggle_ai(AI_ON)
+
+/mob/living/simple_animal/hostile/cryptguard/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
+	if(woke == FALSE)
+		return
+	..()
+
+/mob/living/simple_animal/hostile/cryptguard/sentience_act()
+	awaken() //no infinite health sentient anything oh my god
+
+/obj/effect/temp_visual/cryptguard_death
+	name = "ghostly creature"
+	desc = "there he goes!"
+	icon = 'icons/mob/mob.dmi'
+	icon_state = "ghost_yellow"
+	duration = 30
+
+/mob/living/simple_animal/hostile/cryptguard/leader
+	name = "giant statue"
+	desc = "An incredibly lifelike stone carving depicting a large battle machine."
+	icon_state = "antikytherasword-sleeping"
+	icon = 'icons/mecha/mecha.dmi'
+	sentience_type = SENTIENCE_BOSS
+	loot = list(/obj/mecha/combat/stone/loaded)
+	var/quest = FALSE
+	var/list/guards = list()
+	var/obj/structure/necropolis_gate/trapgate = list()
+
+/mob/living/simple_animal/hostile/cryptguard/leader/Initialize()
+	. = ..()
+	for(var/mob/living/simple_animal/hostile/cryptguard/statues in orange(10, get_turf(src)))
+		guards += statues
+	for(var/obj/structure/necropolis_gate/gates in range(10, get_turf(src)))
+		trapgate += gates
+
+/mob/living/simple_animal/hostile/cryptguard/leader/AttackingTarget()
+	swiping = TRUE
+	var/turf/T = get_turf(target)
+	var/turf/src_turf = get_turf(src)
+	var/dir_to_target = get_dir(src_turf, get_turf(target))
+	var/static/list/front_angles = list(0, -45, 45)
+	var/static/list/side_angles = list(90, -90)
+	var/static/list/back_angles = list(180, 135, -135)
+	visible_message("<span class='warning'>[src] unleashes a flurry of blades!</span>", "<span class='notice'>You begin to unleash a flurry of blades!</span>")
+	var/atktype = pick("cleave", "lunge", "whirlwind", "bdance", "surge")//well good luck
+	switch(atktype)
+		if("lunge")//attacks 3 tiles forward
+			var/turf/Tstep = get_step(T, dir)
+			var/turf/Tstepstep = get_step(Tstep, dir)
+			new /obj/effect/temp_visual/attackwarn(T, src)
+			new /obj/effect/temp_visual/attackwarn(Tstep, src)
+			new /obj/effect/temp_visual/attackwarn(Tstepstep, src)
+		if("cleave") //attacks 3 in front
+			for(var/i in front_angles)
+				var/turf/Tcleave = get_step(src_turf, turn(dir_to_target, i))
+				new /obj/effect/temp_visual/attackwarn(Tcleave, src)
+		if("whirlwind") //attacks angles except diagonals
+			var/static/list/whirlwind_angles = list(0, -90, 90, 180)
+			for(var/i in whirlwind_angles)
+				var/turf/Twhirl = get_step(src_turf, turn(dir_to_target, i))
+				new /obj/effect/temp_visual/attackwarn(Twhirl, src)
+		if("bdance") //attacks diagonals
+			var/static/list/bdance_angles = list(45, -45, 135, -135)
```

Comment:
uh, you seem to be missing something here
also honestly I'd combine whirlwind and bladedance into one attack, randomly picking cardinal/diagonal